### PR TITLE
RNA wdl, inputs

### PR DESCRIPTION
**Pull request by ruchim**

### CODE REVIEW

**Comment by bshifaw** [comment]
is it fine to leave the reference files pointed to gs://rm-dev/refs/ or should they all point to gs://broad-references ?

**Comment by vdauwera** [comment]
Should go to broad references please

File: rna-germline-variant-calling.wdl [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+  	File ref_fasta_index
+
+  	Int preemptible_count
+
+	command {
+		java -jar /usr/gitc/GATK35.jar \
+		-T HaplotypeCaller \
+		-R ${ref_fasta} \
+		-I ${input_bam} \
+		-ERC GVCF \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: don't create gvcfs anymore (and skip GenotypeGVCF in workflow above)

File: rna-germline-variant-calling.wdl [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+		-L ${interval_list} \
+		-dontUseSoftClippedBases \
+		-stand_call_conf 20.0 \
+		-o ${base_name}.vcf.gz \
+		-variant_index_parameter 128000 \
+		-variant_index_type LINEAR
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: don't need above 2 lines

File: rna-germline-variant-calling.wdl [hunk 3]
```diff
@@ -0,0 +1,760 @@
...
+
+		## Is this required?
+		##--max_alternate_alleles 3 \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: No, it's not -- you can delete the above lines

File: rna-germline-variant-calling.wdl [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
+	call MergeVCFs {
+    	input:
+      		input_vcfs = HaplotypeCaller.output_gvcf,
+      		input_vcfs_indexes = HaplotypeCaller.output_gvcf_index,
+      		output_vcf_name = sampleName + ".g.vcf.gz",
+      		preemptible_count = preemptible_count
+  	}
+
+  	call GenotypeGVCFs {
+  		input:
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Let's go directly to VariantFiltration instead

File: rna-germline-variant-calling.wdl [hunk 5]
```diff
@@ -0,0 +1,760 @@
...
+  			ref_fasta_index = refFastaIndex,
+  			ref_dict = refDict,
+  			interval_list = wgsCallingIntervalList,
+  			preemptible_count = preemptible_count
+  	}
+
+	call VariantFiltration {
+		input:
+			input_vcf = GenotypeGVCFs.output_vcf,
+			input_vcf_index = GenotypeGVCFs.output_vcf_index,
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Should now be MergeVCFs.output_vcf

File: rna-germline-variant-calling.wdl [hunk 6]
```diff
@@ -0,0 +1,760 @@
...
+	File interval_list
+
+  	File ref_dict
+  	File ref_fasta
+  	File ref_fasta_index
+
+  	Int preemptible_count
+
+	command {
+		java -jar /usr/gitc/GATK35.jar \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Can we try gatk4 please?

File: rna-germline-variant-calling.wdl [hunk 7]
```diff
@@ -0,0 +1,777 @@
...
+	File inputBam
+	String sampleName = basename(inputBam,".bam")
+
+	File refFasta
+	File refFastaIndex
+	File refDict
+	
+	File wgsCallingIntervalList
+	Int? haplotypeScatterCount = 6
+	Int breakBandsAtMultiplesOf = 100000
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Please remove this and all usages down below

File: rna-germline-variant-calling.inputs.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  "RNAseq.knownVcfsIndices": [
+    "gs://broad-references/Homo_sapiens_assembly19_1000genomes_decoy/Mills_and_1000G_gold_standard.indels.b37.sites.vcf.idx",
+    "gs://broad-references/Homo_sapiens_assembly19_1000genomes_decoy/Homo_sapiens_assembly19_1000genomes_decoy.known_indels.vcf.idx"
+  ],
+  "RNAseq.annotationsGTF": "gs://rm-dev/refs/star/gencode.v19.transcripts.patched_contigs.gtf",
+  "RNAseq.refFasta": "gs://broad-references/Homo_sapiens_assembly19_1000genomes_decoy/Homo_sapiens_assembly19_1000genomes_decoy.fasta",
+  "RNAseq.refFastaIndex": "gs://broad-references/Homo_sapiens_assembly19_1000genomes_decoy/Homo_sapiens_assembly19_1000genomes_decoy.fasta.fai",
+  "RNAseq.refDict": "gs://broad-references/Homo_sapiens_assembly19_1000genomes_decoy/Homo_sapiens_assembly19_1000genomes_decoy.dict",
+  "RNAseq.wgsCallingIntervalList": "gs://rm-dev/refs/decoy/Homo_sapiens_assembly19_1000genomes_decoy.whole_genome.interval_list",
+  "RNAseq.inputBam": "File"
```
Comment (eitanbanks) [inline rna-germline-variant-calling.inputs.json]: please point to test bam

File: rna-germline-variant-calling.wdl [hunk 9]
```diff
@@ -0,0 +1,777 @@
...
+
+	## Requirements for STAR
+	Int? readLength
+	File? zippedStarReferences
+	File annotationsGTF
+  
+  	## Optional user optimizations
+    Int? maxReadsForRealignment
+    Int? compression_level
+    Int? maxRecordsForMergeBamAlignment
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: some of these variables aren't used and should be removed

File: rna-germline-variant-calling.wdl [hunk 10]
```diff
@@ -0,0 +1,777 @@
...
+    File input_bam
+    File input_bam_index
+    String base_name
+    File recalibration_report
+
+    File ref_dict
+    File ref_fasta
+    File ref_fasta_index
+    #Is there a good default for compression level? just remove the -D arg completely if no preference is provided by the user?
+    Int? compression_level
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: I would just remove this and its usage down below

File: rna-germline-variant-calling.wdl [hunk 11]
```diff
@@ -0,0 +1,777 @@
...
+
+	Int preemptible_count
+
+	command <<<
+	 	java -jar /usr/gitc/picard.jar \
+	 	    SortSam \
+	 	    INPUT=${unsorted_bam} \
+	 	    OUTPUT=${base_name}.bam \
+	 	    SORT_ORDER=${sort_order} \
+	 	    COMPRESSION_LEVEL=0 \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: (Do keep this line -- it's the other "compressions" that should be removed and we should use default)
However, you should change this to level=1.

File: rna-germline-variant-calling.wdl [hunk 12]
```diff
@@ -0,0 +1,777 @@
...
+		File recalibrated_bam = ApplyBQSR.output_bam
+		File recalibrated_bam_index = ApplyBQSR.output_bam_index
+		File merged_gvcf = MergeVCFs.output_vcf
+		File merged_gvcf_index = MergeVCFs.output_vcf_index
+		File variant_filtered_gvcf = VariantFiltration.output_vcf
+		File variant_filtered_gvcf_index = VariantFiltration.output_vcf_index
+	}
+}
+
+task SortSam {
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: unused task

File: rna-germline-variant-calling.wdl [hunk 13]
```diff
@@ -0,0 +1,777 @@
...
+	 	## OUTPUT_PER_RG=true \
+	 	## COMPRESS_OUTPUTS_PER_RG=true 
+
+	 	##RE_REVERSE=true \ #Removed from the Firehose workflow, as it's a default
+	 	
+	 	# These are args I've seen recommended by other instances
+	 	# of RNA workflows -- not sure if they apply here.
+	 	## NON_PF=true \ 
+	 	## CLIPPING_ATTRIBUTE=XT \
+      	## CLIPPING_ACTION=2 \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: just remove all these lines

File: rna-germline-variant-calling.wdl [hunk 14]
```diff
@@ -0,0 +1,777 @@
...
+            -XX:+PrintGCDetails -Xloggc:gc_log.log \
+            -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Dsamjdk.compression_level=${default=2 compression_level} -Xms3000m" \
+            ApplyBQSR \
+            --addOutputSAMProgramRecord \
+            -R ${ref_fasta} \
+            -I ${input_bam} \
+            --useOriginalQualities \
+            -O ${base_name}.bam \
+            -bqsr ${recalibration_report} \
+            --EOQ true
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: No!  Please remove.

File: rna-germline-variant-calling.wdl [hunk 15]
```diff
@@ -0,0 +1,777 @@
...
+		    -T HaplotypeCaller \
+		    -R ${ref_fasta} \
+		    -I ${input_bam} \
+		    -ERC GVCF \
+		    -L ${interval_list} \
+		    -dontUseSoftClippedBases \
+		    -stand_call_conf 20.0 \
+		    -o ${base_name}.vcf.gz \
+		    -variant_index_parameter 128000 \
+		    -variant_index_type LINEAR
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Remove above 2 lines

File: rna-germline-variant-calling.wdl [hunk 16]
```diff
@@ -0,0 +1,777 @@
...
+  	File ref_fasta_index
+
+  	Int preemptible_count
+
+	command <<<
+		java -jar /usr/gitc/GATK35.jar \
+		    -T HaplotypeCaller \
+		    -R ${ref_fasta} \
+		    -I ${input_bam} \
+		    -ERC GVCF \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: remove above line

File: rna-germline-variant-calling.wdl [hunk 17]
```diff
@@ -0,0 +1,777 @@
...
+		    -L ${interval_list} \
+		    -dontUseSoftClippedBases \
+		    -stand_call_conf 20.0 \
+		    -o ${base_name}.vcf.gz \
+		    -variant_index_parameter 128000 \
+		    -variant_index_type LINEAR
+
+		## Is this required?
+		##--max_alternate_alleles 3 \
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: remove above 2 lines

File: rna-germline-variant-calling.wdl [hunk 18]
```diff
@@ -0,0 +1,777 @@
...
+
+    runtime {
+        disks: "local-disk 1 SSD"
+        memory: "2 GB"
+        docker: "broadinstitute/genomes-in-the-cloud:2.3.1-1504795437"
+        preemptible: preemptible_count
+    }
+}
+
+task GenotypeGVCFs {
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: remove task

File: rna-germline-variant-calling.wdl [hunk 19]
```diff
@@ -0,0 +1,777 @@
...
+        	SORT_ORDER=${sort_order}
+    >>>
+
+    output {
+        File output_bam = "${base_name}.bam"
+    }
+
+    runtime {
+        docker: "broadinstitute/genomes-in-the-cloud:2.3.1-1504795437"
+        disks: "local-disk " + sub(((size(input_bam,"GB")+1)*5),"\\..*","") + " SSD"
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: Should be HDD

File: rna-germline-variant-calling.wdl [hunk 20]
```diff
@@ -0,0 +1,777 @@
...
+
+	output {
+		File fastq1 = "${base_name}.1.fastq.gz"
+    	File fastq2 = "${base_name}.2.fastq.gz"
+	}
+
+	runtime {
+		docker: "broadinstitute/genomes-in-the-cloud:2.3.1-1504795437"
+		memory: "4 GB"
+		disks: "local-disk " + sub(((size(unmapped_bam,"GB")+1)*5),"\\..*","") + " SSD"
```
Comment (eitanbanks) [inline rna-germline-variant-calling.wdl]: change all SSD to HDD

File: rna-germline-variant-calling.inputs.json (hunk 8 above)
Comment (ruchim) [reply rna-germline-variant-calling.inputs.json]: @eitanbanks 1) I see that gs://broad-public-datasets/NA12878 has a list of unmapped bams for NA12878. This workflow can currently only accept a single input bam and I don't see an aggregated bam for NA12878. 2) All the data in broad-public-datasets seems to be whole genome. Is there someplace where I can find RNA data, or does it not matter?

File: rna-germline-variant-calling.wdl (hunk 7 above)
Comment (ruchim) [reply rna-germline-variant-calling.wdl]: done

File: rna-germline-variant-calling.wdl (hunk 9 above)
Comment (ruchim) [reply rna-germline-variant-calling.wdl]: done

File: rna-germline-variant-calling.wdl (hunk 16 above)
Comment (ruchim) [reply rna-germline-variant-calling.wdl]: re-running without GenotypeGVCFs in FireCloud now.

**Review by ruchim — COMMENTED** [review]
Re-running on test data with these changes to the workflow.

File: rna-germline-variant-calling.inputs.json (hunk 8 above)
Comment (eitanbanks) [reply rna-germline-variant-calling.inputs.json]: No, you need RNA data.  Where did you get the NA12878 RNA data you ran last week?

Comment (ruchim) [reply rna-germline-variant-calling.inputs.json]: I got some NA12878 fastq's from the KCO group

**Comment by bshifaw** [comment]
Ready for merge?

**Comment by ruchim** [comment]
Added public RNA test data to the inputs json and tested in FireCloud.

**Comment by ruchim** [comment]
@bshifaw ready for merge!

**Pull request merged**